### PR TITLE
Clean up the spec docs

### PR DESCRIPTION
**Pull request by olemarkus**
* try to make cluster and IG docs more consistent
* split the IG docs between the spec and common operational use cases

### CODE REVIEW

File: docs/cluster_spec.md [hunk 1]
```diff
@@ -742,9 +744,9 @@ spec:
       image: busybox
 ```
 
-### fileAssets
+## fileAssets
 
-FileAssets is an alpha feature which permits you to place inline file content into the cluster and instanceGroup specification. It's designated as alpha as you can probably do this via kubernetes daemonsets as an alternative.
+FileAssets permits you to place inline file content into the cluster and instanceGroup specification. This is useful for depoying additionl configuration files that kube components requires, such as auditlogs or admission controller configurations.
```
Comment (rifelpet) [inline docs/cluster_spec.md:749]:
```suggestion
FileAssets permits you to place inline file content into the cluster and instanceGroup specification. This is useful for deploying additional configuration files that kubernetes components requires, such as auditlogs or admission controller configurations.
```

File: docs/instance_groups.md [hunk 2]
```diff
@@ -1,452 +1,25 @@
-# Instance Groups
+# The `InstanceGroup` resource
 
-kops has the concept of "instance groups", which are a group of similar machines.  On AWS, they map to
-an AutoScalingGroup.
+The `InstanceGroup` resource represents a group of similar machines typically provisioned in the same availability zone. On AWS, instance groups map directly to an autoscale group.
```
Comment (rifelpet) [inline docs/instance_groups.md:3]:
```suggestion
The `InstanceGroup` resource represents a group of similar machines typically provisioned in the same availability zone. On AWS, instance groups map directly to an autoscaling group.
```

File: docs/instance_groups.md [hunk 3]
```diff
@@ -596,3 +140,36 @@ spec:
 ```
 
 which would end up in a drop-in file on nodes of the instance group in question.
+
+## mixedInstancePolicy (AWS Only)
```
Comment (rifelpet) [inline docs/instance_groups.md:144]: Perhaps an example yaml snippet that contains each of the fields mentioned below?

Comment (olemarkus) [reply docs/instance_groups.md:144]: I can add that. Mind if I do it in a later PR?

Comment (rifelpet) [reply docs/instance_groups.md:144]: yea that's fine with me

**Pull request merged**